src/uct/ib/base/ib_device.h

```diff
@@ -48,15 +49,18 @@ enum {
 enum {
     UCT_IB_DEVICE_FLAG_MLX4_PRM = UCS_BIT(1),   /* Device supports mlx4 PRM */
     UCT_IB_DEVICE_FLAG_MLX5_PRM = UCS_BIT(2),   /* Device supports mlx5 PRM */
-    UCT_IB_DEVICE_FLAG_DC       = UCS_BIT(3)    /* Device supports DC */
+    UCT_IB_DEVICE_FLAG_DC       = UCS_BIT(3),   /* Device supports DC */
+    UCT_IB_DEVICE_FLAG_ONLY_IB  = UCS_BIT(4)    /* Require only IB */
 };
 
 
 typedef enum {
-    UCT_IB_ADDRESS_SCOPE_LINK_LOCAL,   /* Subnet-local address */
-    UCT_IB_ADDRESS_SCOPE_SITE_LOCAL,   /* Site local, 16-bit subnet prefix */
-    UCT_IB_ADDRESS_SCOPE_GLOBAL        /* Global, 64-bit subnet prefix */
-} uct_ib_address_scope_t;
+    UCT_IB_ADDRESS_TYPE_LINK_LOCAL,   /* Subnet-local address */
+    UCT_IB_ADDRESS_TYPE_SITE_LOCAL,   /* Site local, 16-bit subnet prefix */
+    UCT_IB_ADDRESS_TYPE_GLOBAL,       /* Global, 64-bit subnet prefix */
+    UCT_IB_ADDRESS_TYPE_ETH_LOCAL,    /* RoCE - Subnet-local address */
+    UCT_IB_ADDRESS_TYPE_ETH_GLOBAL    /* RoCE - Subnet-global address */
+} uct_ib_address_type_t;
```

Comment:
As I understand, the order if the enum elements must be unchanged. Overwise, the handling algorithm will be broken. Perhaps, it might be helpful to add warning about this.